docs/content/guides/developer/app-examples/trustless-swap.mdx

```diff
@@ -1,27 +1,1957 @@
+</details>
+
+
+If the second party stops responding, the first party can unlock their object. You need to create a function so the second party can recover their object in the symmetric case as well.
+
+- It needs to check that the caller matches `sender`, because `Escrow` objects are shared and anybody can access them.
+- It accepts the `Escrow` by value so that it can clean it up after extracting the `escrowed` object, reclaiming the storage rebate for the sender and cleaning up an unused object on chain.
+
+<details>
+<summary>
+`return_to_sender` function in `shared.move`
+</summary>
+{@inject: examples/trading/contracts/escrow/sources/shared.move#fun=return_to_sender noComments}
+</details>
+
+Finally, you need to add a function to allow the first party to complete the trade.
+
+- This function also accepts the `Escrow` by value because it will consume it after the swap is complete.
+- It checks that the sender of the transaction is the intended recipient (the first party), and that the ID of the key that they provided matches the key specified when the object was escrowed. This ensures no tampering occurs, because this key can be provided only if it had not been used to unlock the object, which proves the object has not left its `Locked<U>` between the call to `create` and to `swap`. You can inspect the `lock` module to see that it cannot be modified while in there.
+- The call to `unlock` further checks that the key matches the locked object that was provided.
+- Instead of transferring the escrowed object to the recipient address, it is returned by the `swap` function. You can do this because you checked that the transaction sender is the recipient, and it makes this API more composable. Programmable transaction blocks (PTBs) provide the flexibility to decide whether to transfer the object as it is received or do something else with it.
+
+<details>
+<summary>
+`swap` function in `shared.move`
+</summary>
+{@inject: examples/trading/contracts/escrow/sources/shared.move#variable=EMismatchedSenderRecipient,EMismatchedExchangeObject singleSpace noComments}
+{@inject: examples/trading/contracts/escrow/sources/shared.move#fun=swap noComments noTitle}
+</details>
+
+
+:::tip Additional resources
+
+- [Full source code](https://github.com/MystenLabs/sui/blob/705ee1ed3ce8cfadc4597c6facb6769d7dfb5896/examples/trading/contracts/escrow/sources/shared.move)
+- Concepts: [Shared Objects](../../../concepts/object-ownership/shared)
+- Concepts: [Shared Object Deletion](https://blog.sui.io/ephemeral-shared-objects/)
+- Concepts: [PTBs](../../../concepts/transactions/prog-txn-blocks)
+
+:::
+
+### Testing
+
+Tests for the `escrow` module will be more involved than for `lock` – you will take advantage of `test_scenario`'s ability to simulate multiple transactions from different senders, and interact with shared objects.
+
+You will focus on the test for a successful swap, but a link to all the tests can be found later on.
```

Comment:
This scans strangely for me -- it comes from changing "We" to "You" but leaving the rest of the sentence unchanged, I think.